(SP:) Create a starting "Про Фундацію" block

lizabre opened the pull request:
Created About foundation block

- implemented tests
- design based on Figma
- starters for responsive design
- added translations

![image](https://github.com/user-attachments/assets/606957ef-92aa-4bb8-a3b6-b89de2a9401f)
![image](https://github.com/user-attachments/assets/4401fe59-982d-4270-a656-6af3c1a1852e)

lizabre committed:
feat: start working on a block

lizabre committed:
feat: about us section

rebase

rebase

lizabre committed:
fix: updated grid tests

lizabre committed:
fix: updated tests

rebase

rebases

lizabre committed:
feat: added translations and tests

rebase

lizabre committed:
fix: updated

Mav-Ivan commented on app/[lang]/layout.tsx:
```diff
@@ -6,12 +6,12 @@ import { notFound } from 'next/navigation';
 import ThemeProvider from '~/ds-components/theme/ThemeProvider';
 
 import { routing } from '~/i18n/routing';
+import Footer from '~/shared/components/Footer/Footer';
```
```suggestion
import Footer from '~/components/Footer/Footer';
```

We have aliases now, don`t forget to use them

lizabre replied:
Thank you! Updated those

Mav-Ivan commented on app/[lang]/page.tsx:
```diff
@@ -3,12 +3,14 @@ import React from 'react';
 import LiatoshynskyOffice from '~/components/Liatoshynsky-office/LiatoshynskyOffice';
 
 import FoundationFounders from '~/shared/components/FoundationFounders/FoundationFounders';
+import AboutFoundation from '~/shared/components/main-page-sections/about-foundation/AboutFoundation';
```
```suggestion
import FoundationFounders from '~/components/FoundationFounders/FoundationFounders';
import AboutFoundation from '~/components/main-page-sections/about-foundation/AboutFoundation';
```

Mav-Ivan commented on app/shared/components/main-page-sections/about-foundation/AboutFoundation.styles.ts:
```diff
@@ -0,0 +1,182 @@
+import { Theme } from '@mui/material';
+
+import { theme } from '../../design-system/all-components/theme/Theme';
```
```suggestion
import { theme } from '~/ds-components/theme/Theme';
```

Mav-Ivan commented on app/shared/components/main-page-sections/about-foundation/AboutFoundation.tsx:
```diff
@@ -0,0 +1,85 @@
+import Image from 'next/image';
+import { getTranslations } from 'next-intl/server';
+
+import { theme } from '../../design-system/all-components/theme/Theme';
```
```suggestion
import { theme } from '~/ds-components/theme/Theme';
```

uliaescha commented on app/[lang]/layout.tsx:
```diff
@@ -55,40 +55,37 @@ export default async function RootLayout({ children, params }: RootLayoutParams)
-              </Container>
-              <Footer />
-            </Box>
+            <Container
```
seems that you overwrote already merged code and the old footer issue appears again

lizabre replied:
You're right, missed it
Now it's here

lizabre committed:
fix: reviewed footer

rebase

lizabre committed:
fix: updated aliases

rebase

Iarynovskyi commented on app/[lang]/page.tsx:
```diff
@@ -3,12 +3,14 @@ import React from 'react';
-      <LiatoshynskyOffice />
+      <AboutFoundation />
       <FoundationFounders />
+      <LiatoshynskyOffice />
```
```suggestion
      <AboutFoundation />
       <LiatoshynskyOffice />
      <FoundationFounders />
```
If I remember correctly, the order in figma is like this

lizabre replied:
updated with new added section as well

Iarynovskyi commented on app/shared/components/design-system/all-components/theme/Theme.ts:
```diff
@@ -20,7 +20,7 @@ export const theme = createTheme({
       md: 1024,
       sm: 768,
-      xs: 0
+      xs: 376
```
Is this a must change now ?
just all other styles will float now if you do this

https://github.com/user-attachments/assets/9ac24e7c-98dc-47e7-9ff5-158f990d1240

lizabre replied:
my fault, thanks for noticing, now it's back

Iarynovskyi commented on public/images/foundation-main.png:
Do rebase with develop and than delete this photo and add them again, i fixed that bug
Because now they broken here

lizabre replied:
done for both

Iarynovskyi commented on public/images/liatoshynsky-main-photo.png:
With this photo to

lizabre committed:
fix: updated from comments

rebase

lizabre committed:
fix: some fixes

lizabre committed:
fix: updated photos and some styles

lizabre committed:
fix: tests again

Iarynovskyi commented:
After `xs` changing, your styles have float a bit

lizabre commented:
> After `xs` changing, your styles have float a bit

I've only noticed the issue with photo at the end and image caption, could you please point out the others, if they're present?

lizabre committed:
fix: style fixes

VKormylo commented on app/shared/components/main-page-sections/about-foundation/AboutFoundation.tsx:
```diff
@@ -0,0 +1,86 @@
+    <Box sx={sectionStyles.conatiner}>
+      <Typography sx={sectionStyles.title}>{title}</Typography>
+      <Box sx={sectionStyles.photoContainer}>
+        <Box sx={sectionStyles.yellowBlock} />
```
You can also add border prop to ImageWithCaption which takes sizes and left and top coords, and achieve that yellow border. Take a look at that component.

Iarynovskyi commented:
https://github.com/user-attachments/assets/3dad0f0d-ad5d-476c-8ffc-df728edbdee3

Problems only with adaptability.
If we are not going to show it in this demo then all fine

VKormylo commented:
> > 2025-06-11.17.32.15.mov
> > Problems only with adaptability. If we are not going to show it in this demo then all fine
> 
> Oh, I see, thanks I'll try in the nearest half an hour to make it work, if i'd have to create a new task for it - I'll let you know

I updated ImageWithCaption a little bit in last PR, so make a rebase and take a look. You should wrap that image in some grid or flex, and then it shouldn't overflow. Also there is align prop now for caption alignment.

lizabre committed:
fix: yellow block update

lizabre committed:
fix: style

lizabre committed:
fix: styles

lizabre commented:
> 2025-06-11.17.32.15.mov 
> Problems only with adaptability. If we are not going to show it in this demo then all fine

I've tried to resume as much as I could for my last commit, thank you once again

lizabre commented:
> > > 2025-06-11.17.32.15.mov
> > > Problems only with adaptability. If we are not going to show it in this demo then all fine
> > 
> > 
> > Oh, I see, thanks I'll try in the nearest half an hour to make it work, if i'd have to create a new task for it - I'll let you know
> 
> I updated ImageWithCaption a little bit in last PR, so make a rebase and take a look. You should wrap that image in some grid or flex, and then it shouldn't overflow. Also there is align prop now for caption alignment.

It was my issue actually with margins, thanks for your help!
Also updated the visual yellow block
